network/src/application/management.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    application::{db::PeerDb, interface::PeerStateChange, types::PeerError},
+    transport::ConnectionMetadata,
+};
+use diem_types::PeerId;
+use std::sync::Arc;
+
+/// An interface to manage peers being added and removed from the `PeerDb`.
+/// Meant to be handled by `PeerManager`
+/// TODO: Handle connections and peers separate so we can have multiple connections per peer
+pub trait PeerManagementInterface {
+    fn db(&self) -> &PeerDb;
```

Comment:
it seems every trait exposes the underlying db, are they all necessary? seems a leaky abstraction

Reply:
I've cut some of these.  I was trying to not repeat code, while preventing certain implementations from changing state / disconnecting etc.  I agree it needed to be improved